src/Symfony/Component/Routing/Generator/AstGenerator/GeneratorAstGenerator.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Generator\AstGenerator;
+
+use PhpParser\BuilderFactory;
+use PhpParser\Comment;
+use PhpParser\Node\Name;
+use PhpParser\Node\Param;
+use PhpParser\Node\Stmt;
+use PhpParser\Node\Expr;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Ast\NodeList;
+use Symfony\Component\Ast\Util\AstHelper;
+use Symfony\Component\Routing\Exception\RouteNotFoundException;
+use Symfony\Component\Routing\Generator\UrlGenerator;
+use Symfony\Component\Routing\RequestContext;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * @author Guilhem N. <egetick@gmail.com>
+ */
+final class GeneratorAstGenerator implements GeneratorAstGeneratorInterface
+{
+    public function __construct(RouteCollection $routes)
+    {
+        $this->routes = $routes;
+        $this->factory = new BuilderFactory();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function generate(array $options = array())
```

Comment:
I reused the current signature. I have no strong opinion about passing the route collection but it looks weird to me to not know the name of the class that will be generated (most of the time the caller doesn't instantiate the class).